### ISSUE TITLE
Better name for QPath

### ISSUE THREAD

**Issue by matklad** [rust-lang-nursery/wg-grammar#66]
Reverse-turbofish paths like `<_>::default` are currently known as qualified paths, or QPaths. I've always found this terminology pretty confusing, because a `type` doesn't seem like a more qualifierer qualifier than `crate::` or `super::`. I wonder if there's a better name for this production... Or are other folks happy with this naming?

**Comment by eddyb** [comment]
They're called type-qualified paths, the turbofish name is entirely unofficial.
And yes, a type is far more significant than just a path keyword.

The type *isn't a namespace* in a type-qualified path, it's the type to search for an associated item on. And such a path is resolved during/after type inference, as opposed to during name resolution.

**Comment by matklad** [comment]
Yeah, the important word in “type qualified path” is the *type*, but QPath picks the less important :(

**Comment by Centril** [comment]
Call it `TQPath` ?

**Comment by eddyb** [comment]
QPath is from rustc and it's a shorthand for MaybeQualifiedPath.
I guess in the grammar we could expand it out.

There's also "fully-qualified" or "trait-qualified" paths (`<T as Trait>::...`) which are also QPath.
